src/job.ts

```diff
@@ -169,7 +169,9 @@ export class Job {
         predefinedVariables["CI_ENVIRONMENT_NAME"] = this.environment?.name ?? "";
         predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").slice(0, 23).replace(/-$/, "").toLowerCase() ?? "";
         predefinedVariables["CI_ENVIRONMENT_URL"] = this.environment?.url ?? "";
-        predefinedVariables["CI_NODE_INDEX"] = `${opt.nodeIndex}`;
+        if (opt.nodeIndex !== null) {
+            predefinedVariables["CI_NODE_INDEX"] = `${opt.nodeIndex}`;
+        }
```

Comment:
I tested on gitlab.com, and the variable is not set (i.e. it is not present in the environment, not set to an empty string).

Looking at the code around this though, it seems the other predefined variables are set to an empty string instead. I don't care either way, but I just wanted to point this out (since it breaks the surrounding code style).

Reply:
seems fine to me. 

But for regression testing, could you add a simple test case for this bug fix ?